pkg/volume/csi/csi_plugin.go

```
@@ -620,10 +631,45 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 		}
 		return false, err
 	}
-	if csiDriver.Spec.AttachRequired != nil && *csiDriver.Spec.AttachRequired == false {
-		return true, nil
+
+	skipAttach := false
+	if csiDriver.Spec.AttachRequired != nil {
```

the skip attach field will be defaulted in the API. I don't think we should default it here. We can return an error if it's unset.

---

Will fix.

---

Ok

---

Reverting logic to original code.  AttachedRequired unset has overloaded meaning for older drivers which had no way of specifying.